components/TimeoutModal.tsx

```diff
@@ -5,76 +5,71 @@ import Modal from 'react-bootstrap/Modal'
 import { Button } from './Button'
 import getUrl from '../utils/getUrl'
 
-let timeOutTimerId: NodeJS.Timeout | null = null
 let warningTimerId: NodeJS.Timeout | null = null
 
 export interface TimeoutModalProps {
-  action: string
+  userArrivedFromUioMobile: boolean
   timedOut: boolean
 }
 
 export const TimeoutModal: React.FC<TimeoutModalProps> = (props) => {
   const { t } = useTranslation()
-  const { action, timedOut } = props
-  const TIMEOUT_MS = 30 * 60 * 1000
-  const TIMEOUT_DISPLAY_TIME_IN_MINUTES = 5
-  const TIMEOUT_WARNING_MS = TIMEOUT_MS - TIMEOUT_DISPLAY_TIME_IN_MINUTES * 60 * 1000
-  const [numberOfMinutes, setNumberOfMinutes] = useState(TIMEOUT_DISPLAY_TIME_IN_MINUTES)
-  const [showWarningModal, setShowWarningModal] = useState<boolean | null>(timedOut)
+  const { timedOut, userArrivedFromUioMobile } = props
 
-  useEffect(() => {
-    if (showWarningModal) {
-      const timer = setTimeout(() => {
-        setNumberOfMinutes(numberOfMinutes - 1)
-      }, 60 * 1000)
-      return () => clearTimeout(timer)
-    }
-  })
+  // handy converter for Minutes -> Milliseconds
+  const ONE_MINUTE_MS = 60 * 1000
+
+  // keep times in human readable minutes
+  const REDIRECT_TIMER = 30
+  const WARNING_DURATION = 5
+  const WARNING_TIMER = REDIRECT_TIMER - WARNING_DURATION
```

Comment:
Converted these to human readable minutes, converting to milliseconds just in time